minari/utils.py

```diff
@@ -394,60 +384,139 @@ def create_dataset_from_buffers(
     except InvalidSpecifier:
         print(f"{minari_version} is not a version specifier.")
 
-    if observation_space is None:
-        observation_space = env.observation_space
-    if action_space is None:
-        action_space = env.action_space
+    dataset_metadata["minari_version"] = minari_version
 
     if expert_policy is not None and ref_max_score is not None:
         raise ValueError(
             "Can't pass a value for `expert_policy` and `ref_max_score` at the same time."
         )
 
-    dataset_path = get_dataset_path(dataset_id)
-
-    # Check if dataset already exists
-    if os.path.exists(dataset_path):
-        raise ValueError(
-            f"A Minari dataset with ID {dataset_id} already exists and it cannot be overridden. Please use a different dataset name or version."
+    if eval_env is None:
+        warnings.warn(
+            f"`eval_env` is set to None. If another environment is intended to be used for evaluation please specify corresponding Gymnasium environment (gym.Env | gym.envs.registration.EnvSpec).\
+                  If None the environment used to collect the data (`env={env_spec}`) will be used for this purpose.",
+            UserWarning,
         )
-    dataset_path.mkdir()
-
-    dataset_path = os.path.join(dataset_path, "data")
-    storage = MinariStorage.new(
-        dataset_path,
-        observation_space=observation_space,
-        action_space=action_space,
-        env_spec=env.spec,
-    )
+        eval_env_spec = env_spec
+    else:
+        if isinstance(eval_env, str):
+            eval_env_spec = gym.spec(eval_env)
+        elif isinstance(eval_env, EnvSpec):
+            eval_env_spec = eval_env
+        elif isinstance(eval_env, gym.Env):
+            eval_env_spec = eval_env.spec
```

Comment:
do you think it is helpful to support all these 3 cases?

Reply:
I think it can be useful, I took the idea from `gym.make()`. For example, if you want to save the eval_env as a modified version of the environment used to collect data you can just get the EnvSpec of the environment, update any of the parameters and pass it as the `eval_env` without having to create another `gym.Env`